src/main/scala/ru/butik/butifactory/ApkService.scala

```diff
@@ -11,12 +12,23 @@ case class ApkFileContainer(apkFile: AbstractApkFile, file: File) extends Closea
 }
 
 class ApkService(datastore: Datastore, storageBackend: ArtifactStorageBackend, frontend: ArtifactStorageFrontend, pushService: PushService) {
+  val bufferSize = 8192
 
   def uploadFile(apk: ApkFileContainer): Either[String, ArtifactVersion] = {
     val artifactName = apkToName(apk.apkFile)
     val version = apk.apkFile.getApkMeta.getVersionName
     val versionCode = apk.apkFile.getApkMeta.getVersionCode
 
+    val buffer = new Array[Byte](bufferSize)
+    val md5Instance = MessageDigest.getInstance("MD5")
+    val dis = new DigestInputStream(new FileInputStream(apk.file), md5Instance)
+    try {
+      while (dis.read(buffer) != -1) { }
+    } finally {
+      try { dis.close() } catch { case _: IOException => Left("resource closing exception") } finally {}
```

Comment:
Left is unnecessary while you are not returning value from this block.
You can just use empty block or idiomatic `scala.util.control.Exception.ignoring` like
```
scala.util.control.Exception.ignoring(classOf[ExceptionType]) {
  // suppressed code here
}
```

Also `finally` block is useless if empty.

Reply:
Done, I used empty catch block, but finally should be if I understood correctly. If I will remove finally block - the code below become unreachable.

Reply:
@Pilot1258 `finally` change execution order only if contains uncatched exception

Reply:
It's weird, but IDEA still show that the code below become unreachable, if I remove the finally. It's necessary to remove it?

Reply:
@khamutov If it's okay, approve please) I've fixed marks for Travis

Reply:
It's an Idea problem,  not scala
https://youtrack.jetbrains.com/issue/SCL-13557

also - build not yet passed
pls run `sbt assembly` locally before pushing PR

Reply:
Done, I've fixed the problem with `sbt assembly`.
Also I've removed the `finally` block.